crypto/bn/bn_recp.c

```diff
@@ -15,6 +15,7 @@ void BN_RECP_CTX_init(BN_RECP_CTX *recp)
     bn_init(&(recp->N));
     bn_init(&(recp->Nr));
     recp->num_bits = 0;
+    recp->shift = 0;
```

Comment:
This will invalidate the BN_DEBUG mechanism behind bn_init(), i think ...
BN_RECP_CTX_init is now an internal method, no need to improve it.